ebpf.mk

```diff
@@ -1,6 +1,14 @@
 # Needs to be set because of PT_REGS_PARMx() and any other target
 # specific facility.
-ARCH := x86
+DEFAULT_ARCH := $(patsubst target_arch="%",%,$(filter target_arch="%",$(shell rustc --print cfg)))
+__ARCH := $(if $(CARGO_CFG_TARGET_ARCH),$(CARGO_CFG_TARGET_ARCH),$(DEFAULT_ARCH))
+x86_64 := x86
+aarch64 := arm64
+powerpc64 := powerpc
+s390x := s390
+# Mappings takes precedence over custom ARCH
+ARCH := $(if $($(__ARCH)),$($(__ARCH)),$(__ARCH))
```

Comment:
If that's possible I would invert the logic here, to set `CARGO_CFG_TARGET_ARCH` using `ARCH`. This has two advantages:
- `ARCH` is commonly used in various projects for this.
- We would not rely on a Cargo specific variable and instead both targets (Rust & eBPF) would be abstracted.

Then we could run `ARCH=aarch64 make`.

Reply:
I thought about it but I ended up with this because CARGO one makes it clear (they are documented) what are the arch names (then mapped to "ebpf" ones). 
 
If we go with `ARCH` the names must be documented (help, at least), but we claim (for good reasons) that no support is available yet, so it should not be documented at the same time (essentially we have to do it later).

Just sharing the reasoning that brought to this. I'm fine with the change.

Reply:
> If we go with `ARCH` the names must be documented (help, at least), but we claim (for good reasons) that no support is available yet, so it should not be documented at the same time (essentially we have to do it later).

Yep, IMO that's fine. We'll have to document which ones are supported, so we'll have to list the right values anyway. If an arch is not supported and someone wants to build it, they'll have to look up what's needed on Cargo and BPF side anyway.